src/app/clusters/camera-av-stream-management-server/camera-av-stream-management-server.cpp

```diff
@@ -2050,6 +2050,22 @@ void CameraAVStreamMgmtServer::HandleCaptureSnapshot(HandlerContext & ctx,
     auto & requestedResolution = commandData.requestedResolution;
     ImageSnapshot image;
 
+    VerifyOrReturn(!mAllocatedSnapshotStreams.empty(), {
+        ChipLogError(Zcl, "CameraAVStreamMgmt[ep=%d]: No snapshot streams are allocated", mEndpointId);
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::NotFound);
+    });
+
+    if (!snapshotStreamID.IsNull())
+    {
+        auto found = std::find_if(mAllocatedSnapshotStreams.begin(), mAllocatedSnapshotStreams.end(),
+                                  [&](const SnapshotStreamStruct & s) { return s.snapshotStreamID == snapshotStreamID.Value(); });
+        VerifyOrReturn(found != mAllocatedSnapshotStreams.end(), {
+            ChipLogError(Zcl, "CameraAVStreamMgmt[ep=%d]: No snapshot stream exist by id %d", mEndpointId,
+                         snapshotStreamID.Value());
+            ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::NotFound);
+        });
+    }
+
     VerifyOrReturn(commandData.requestedResolution.width >= 1 && commandData.requestedResolution.height >= 1,
```

Comment:
It looks we are still missing the following two checks per spec, the second one could be offloaded to the delegate

1. If either SoftLivestreamPrivacyModeEnabled or HardPrivacyModeOn is true:
Fail the command with the status code INVALID_IN_STATE, and end processing with no other side effects.

2. If the SnapshotStreamID is null:
Use the closest existing snapshot stream that is greater than or equal to the RequestedResolution size.
If there are no allocated streams greater than or equal to the RequestedResolution:
Use the stream with the closest possible resolution.